Enable ingress mode for Eureka/Consul

rshriram opened the pull request:
**What this PR does / why we need it**:
Ingress is a generic concept applicable across platforms. K8S has ingress as first class resource (except its broken in several ways). In non k8s platforms, we cannot use this ingress resource spec reliably, as it requires enabling ingress controller without the status syncer (adds more complexities), especially if people are just adding it for the dummy ingress.

This PR defines a new proxy agent mode (LB) that allows the sidecar to obtain just outbound configs, and run as standalone envoy [which becomes ingress in Consul/Eureka]. With this, we would no longer have to expose services directly (like the way we do today). An added benefit is that this would allow people to run envoy as standalone using load balancer service in kubernetes, with support for TCP and HTTP services.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:
<!--  Steps to write your release note:
1. Use the release-note-* labels to set the release note state (if you have access)
2. Enter your extended release note in the below block; leaving it blank means using the PR title as the release note. If no release note is required, just write `NONE`.
-->
```release-note
new Load Balancer mode for sidecar to enable ingress functionality in non-k8s environments
```

kyessenov commented on proxy/envoy/config.go:
```diff
@@ -439,9 +443,17 @@ func buildOutboundListeners(mesh *proxyconfig.MeshConfig, sidecar proxy.Node, in
-		listeners = append(listeners,
-			buildHTTPListener(mesh, sidecar, instances, routeConfig, WildcardAddress, port,
-				fmt.Sprintf("%d", port), false, EgressTraceOperation))
+		var l *Listener
```
nit: use variable to pass different trace operation, use remote address flag values (e.g. `useRemoteAddress := true; operation := EgressTraceOperation`)

rshriram commented:
@kyessenov I fixed the TCP listener part.. It will only open listeners with wildcard listeners.. Also, removed support for egress rules from teh router (aka load balancer) as semantics are not really clear. I also found some stray implementation about egress proxy support in the egress rule, and got rid of that as well. 

And renamed the localhost:8888 to something more readable (istio-pilot/mixer.istio-system) so that the goldens are bit more grok-able..
